addons/project/models/project.py

```diff
@@ -1888,6 +1889,11 @@ def _load_records_create(self, vals_list):
                 else:
                     for field_name in self._get_recurrence_fields() + ['recurring_task']:
                         vals.pop(field_name, None)
+            project_id = vals.get('project_id')
+            if project_id:
+                project = self.env['project.project'].browse(project_id)
+                default_val = self.with_context(default_project_id=project.id).default_get(['project_id'])
+                vals.update(**default_val)
         return super()._load_records_create(vals_list)
```

Comment:
What I am afraid of is that:
* we kind of duplicate the logic we already have in default_get
*  `vals.update(**default_val)` will override any existing values
As discussed with @xavierbol it would be best to do:
```suggestion
            project_id = vals.get('project_id')
            if project_id:
                self = self.with_context(default_project_id=project.id)
        return super()._load_records_create(vals_list)
```
Maybe we need to explicitly pass self in the super because of the context change... I am unsure about that, can you test and only add it in the super if required?

Reply:
> 

Done.
FYI-Remove the account.analytic.tag model in this PR https://github.com/odoo/odoo/pull/93554/commits